### ISSUE TITLE
Missed dailies do massive amounts of damage

### ISSUE THREAD

**Issue by mmarkey3** [HabitRPG/habitica#3161]
My party didn't even realize this wasn't normal until we asked the community; we just thought HabitRPG was that hard.  Apparently the game isn't supposed to be like this by design.  But....

My party has been doing quests for a while, and we've been getting extremely frustrated because the rewards have never outweighed the losses from dying 5 times per quest.  We have a level 60 healer who can't keep up with damage done from missed dailies, and I (the only one who hasn't died a bunch) spend 75 GP on potions EVERY DAY.

There are four people in our party, who each have 10-30 dailies.  We each do about 90% of those dailies every day, and end up doing 10-25 damage to the monster every day.  However, we all miss 1-2 dailies per day.  On a good day, everyone does 5HP in damage to themselves and the party - resulting in about 25 HP of damage to each person.  On a GOOD day.  Our healer recently missed one blue and one red daily and did 25 damage to himself and the rest of the party, plus the ~20 damage the two of us did who logged in before our fourth person (lucky she didn't, otherwise we would have wiped).  Every day I go from 50 HP to 5 HP and need to potion back up (thus the 75 GP/day on potions).

It's not like we're not doing a lot of tasks.  I have 75GP to spend on potions every day, and I'm still making a profit.  It's just that each day is a strong possibility of one-hit TPK, even when we're all at max health and it looks like it's going well.

After talking to some others, I discovered that most 7-person parties only take 10-15 damage/day total, on average.  Our average is 35-45 damage/day after only missing 1 daily/person.  Even easy blue dailies do 4-5 damage/day.  What's the deal?

**Comment by deilann** [comment]
Are you positive all of these dailies are registering as checked off? Habit's been having server issues lately, so sometimes it'll check off locally, but not save to the server. This means that when cron rolls around, it'll count those as unchecked. By doing a refresh right before you're done checking things off, you can see if those clicks are actually sticking. (If not, try again. Make sure to click slowly, as clicking faster increases your chance of an error.)

**Comment by mmarkey3** [comment]
We have refreshed in the evening to ensure this and I check it on mobile as well.  I also see the graphs change when I do the items as opposed to not doing them, and they appear accurate.  The only thing I can think of is it might be drawing from local cache, since we don't hard refresh.  But I just recently cleared my whole cache (and checked to make sure all but one of my dailies were checked off) and woke up to the same huge amounts of damage this morning.

One person in our party was having difficulty with cron not happening at one point, but this hasn't affected anyone else, and this problem is affecting the whole party whether his cron occurs or not.

**Comment by deilann** [comment]
Can you show me a ss of your party screen?

**Comment by mmarkey3** [comment]
![partyscr1](https://cloud.githubusercontent.com/assets/5139847/2542646/1df8e6d2-b5ea-11e3-8196-3be114ddd490.JPG)
![partyscr3](https://cloud.githubusercontent.com/assets/5139847/2542647/1df90dd8-b5ea-11e3-92a1-2fe1550f3718.JPG)
![partyscr2](https://cloud.githubusercontent.com/assets/5139847/2542648/1df95a9a-b5ea-11e3-94ae-1372a8171ac5.JPG)

Could let me know if numbers like these are actually normal for 1-2, at most 3 missed dailies?

**Comment by mmarkey3** [comment]
There was also a point in there somewhere where two of us, Blaze and Gene, were in the inn (but we were still taking damage of course).  I was logging in every day during that to quaff potions even though I wasn't doing dailies.

Today I only had the GP for one potion...working on that, lest I die tomorrow....

**Comment by deilann** [comment]
Okay, so, looking at it, I know that there is one thing to keep in mind--
Vice has a STR of 3, so is going to deal more than normal damage for missed
dailies. From my personal experience, I would say that the lower damage
seems not that odd. Vice is supposed to be pretty frickin' tough. It's also
entirely possible that every once in a while, some extra "undone" dailies
are sneaking in, due to the server issues. Here's what I would suggest when
fighting Vice-- as he's dealing super extra damage, if you have red dailies
they either absolutely _need_ to be done or the user needs to rearrange
their dailies so that it's a habit instead.

I'm not saying this is for sure 100% normal, but it's _really_ hard to tell
because I don't know all your task values. But it could be.

**Comment by deilann** [comment]
kk looking at my own damage -- I'm getting between 1 and 2 damage for a blue - habit. So that doesn't seem extraordinary, but may be high, if that makes sense, after the multiplier. It's also good to note that if the tasks are not set easy, but set medium or hard, that will increase the amount of damage they do.

**Comment by mmarkey3** [comment]
So...Vice is doing 3x's the normal amount of damage?  So missing one hard daily at -44 and one easy daily at + 0.02529 will result in 23 damage to both the party and yourself?  That is, missing a hard dark red daily will do 40 points of damage to you by itself?  By design???

**Comment by deilann** [comment]
The stats given by the progress graph aren't exactly the task value. But if
something is seriously that red and on hard, that sounds right. If someone
is missing a daily that consistently, I highly suggest they rethink how
they're doing their dailies and maybe switch it to a habit until they feel
like it's possible for them to do.

Vice is doing 1.5x damage, actually, 'cause a static multiplier is missing.
But Vice also ignores CON.

**Comment by mmarkey3** [comment]
That still seems really high even for a tough fight, considering the entire party can so easily one-hit-KO based on a single person, or even due to a cron error for one person (thankfully not the case here).  That doesn't sit well with me.  But it's not a bug apparently; it's part of the design.  Returning to Trello to discuss further.

**Comment by deilann** [comment]
I think the idea is that if you're in a boss fight, you should not ignore
super red dailies... else you'll be dealing massive amounts of damage to
your party members.

**Comment by mmarkey3** [comment]
My personal red dailies aren't really ignored persay...just failed....
I'll admit some of us are using dailies for the wrong kind of task - like things that can pass you by and you CAN'T complete, such as "get to work on time" or "get to sleep on time" or "don't yell at anyone today" or "make sure you check this news item at this specific time" or "write up/check your daily schedule and resolve the conflicts (useless if not done before your day officially starts, or yknow, after you come home from work)."  The idea is that those should be habits, I understand.  But they also recur daily, only once daily, only on certain days of the week, and should be punished if missed.

I always forget to apply the punishments to my habits, so unless I made a daily to apply the penalties on my habits....  And then there's no way of knowing "Did I actually apply that for today?" because they don't gray out or reset on cron.  Unless I made a daily with a checklist for each habit that should be penalized daily....  Dailies are a better way to keep track of things like that...as long as, you know, you're not in a quest and they're not murdering you and your friends.

I realize that's not the "right" way to do it.... _sigh_  I'll have to re-evaluate how I do everything...and make my buddies do the same...

**Comment by deilann** [comment]
For some people they work well as dailies... and for others not so much.
One thing you could do is have a daily of "punished for bad habits" with a
checklist of those habits. Then, it's hard to forget. When you look at that
daily, you can ask yourself "Did I make it to work on time today? Well,
nope! Click the - habit, check off, move on." Then you have a mark saying
you did, in fact, take the hit.

**Comment by deilann** [comment]
Personally for me they don't work so well because I know once I've missed
one, I _cannot_ get a perfect day. So it's seriously demotivating. :c

**Comment by deilann** [comment]
Hey @MarkVaughn that isn't really constructive. While bosses aren't meant to be easy, they aren't meant to be frustrating to the point that users find them demotivating -- that defeats the purpose. Please refrain from making such comments in the future-- @mmarkey3 is understandably frustrated, and being dismissed is not going to help that frustration, only make it worse.

**Comment by mmarkey3** [comment]
@MarkVaughn I do understand your feeling on that.  Still...it's also not the final boss...only the last boss of that particular quest string.  I only expect not to be 1-hit-TPK'd from missing a single daily when the party is level 60 and equipped in the most munchkin way possible and using all our skills in the most strategically apt way possible and completing 30 tasks each per day.  I don't expect it to be a walk in the park or anything, but you really can't say it's our fault at that point. :)  That's all it is.

I'm only here, besides, because I talked to some other Habiteers who were taking at most 10 damage/person, total, during quests.  We've always taken around 30/person.  It just became unbearable once we hit Vice and we were taking 45/person.

**Comment by mmarkey3** [comment]
@deilann yeah I just made a daily checklist for my habits because that was the only way it was gonna work out that way.

**Comment by deilann** [comment]
I find that meta-tasks can be extraordinarily helpful in managing Habit.
Frex, I have master checklists for things that have their own dailies but
are related (so do all studying or do all housework) and often have a daily
of doing a specific daily or to-do, decided on the night before.

**Comment by mmarkey3** [comment]
Yeah I have a "do one to-do" daily, otherwise I'd just never do my to-dos. (Incidentally, that's how I'm making that 75+ gold/day....  All my todos are red....  It's because I do one and add two more....)

**Comment by Luizammend** [comment]
Hi I have a similar issue, I'm on a easy quest (the Rat King) and I missed two dailies (one green and the other blue) and this translated to 21.7 damage points to me and my party. I checked my dailies, and nothing else was "undone". Also my damage usually take 24 to 48 hs to show on the party board (this since always). And For three times I had perfect days and received damage on my turn (the first time I thought I mischecked something, but on the second and third times I confirmed that this was a bug).

**Comment by deilann** [comment]
It sounds like you're not fetching your most recent party messages if you
think the damage isn't showing for more than a day. You need to refresh the
party page or ask for the new messages to see where the damage is coming
from. Also, the rat king is not an easy quest. It is one of the hardest.
Also remember that you take damage from both the boss and your dailies
directly and you'll be taking damage from your party members.

**Comment by Luizammend** [comment]
The damage I referred to (21.7) came only from my turn.

Luiza Mendonça attacks Rat King for 0.0 damage, Rat King attacks party for
21.7 damage.
 -5 hours ago

What happened previously to me is that my damage takes a long time to show
(like 48 to 72 hs) and I complain to my party mates about that. On the next
day the damage shows below my complain in the timeline, as it was there all
along, but it wasn't... Note below, when I wrote :

It's definitely my damage. Do you see the last time I attacked? It was a
perfect day, and I only gave 4.9 of damage and received 5.4! Sometimes I
get the impression that those attacks are terribly delayed, like 48 or 72
hs... Is it just me?
 -18 days ago+1
 The last damage I had done was 4.9/5.4, then on the next day only, the
line :

   -

   Luiza Mendonça attacks Rat King for 21.8 damage, Rat King attacks party
   for 0.0 damage.
    -18 days ago+1

appeared below my complain.

   -

   I'll do that ferox, thanks
    -17 days ago+1
   Luiza Mendonça
   -

   Yeah, it's definitely a bug. Yesterday when I wrote you guys my last
   attack was 4.9/5.4, and now, after I sent you guys the message complaining
   about that, it says it was 21.8/0 (and is before my message in the
   scrolling, as it was there before, but it wasn't). Am I sounding crazy?
    -17 days ago+1
   Luiza Mendonça
   -

   annemarie3013 attacks Rat King for 15.4 damage, Rat King attacks party
   for 0.0 damage.
    -17 days ago+1
   -

   Luiza Mendonça attacks Rat King for 7.1 damage, Rat King attacks party
   for 0.0 damage.
    -17 days ago+1
   -

   Maeve Autumn attacks Rat King for 5.8 damage, Rat King attacks party for
   0.0 damage.
    -17 days ago+1
   -

   Cody Weston attacks Rat King for 10.6 damage, Rat King attacks party for
   0.0 damage.
    -18 days ago+1
   -

   @Luiza Mendonça It might be worthwhile to take a screencap of your task
   list after everything has been ticked off, and then a screencap of the
   damage done, so you can let the developers know about it.
    -18 days ago+1
   ferox
   -

   ferox casts Valorous Presence for the party.
    -18 days ago+1
   -

   @Luiza Mendonça Attacks aren't delayed. Sounds like you've got a real
   bug going on there. :(
    -18 days ago+1
   ferox
   -

   It's definitely my damage. Do you see the last time I attacked? It was a
   perfect day, and I only gave 4.9 of damage and received 5.4! Sometimes I
   get the impression that those attacks are terribly delayed, like 48 or 72
   hs... Is it just me?
    -18 days ago+1
   Luiza Mendonça
   -

   Luiza Mendonça attacks Rat King for 21.8 damage, Rat King attacks party
   for 0.0 damage.
    -18 days ago+1
   -

   Cody Weston attacks Rat King for 38.4 damage, Rat King attacks party for
   0.0 damage.
    -18 days ago+1
   -

   Maeve Autumn attacks Rat King for 5.8 damage, Rat King attacks party for
   0.0 damage.
    -18 days ago+1
   -

   @Luiza Mendonça Yeah you're only supposed to get damage from undone
   dailies Unless it's freaking out about when you're checking your boxes, it
   might be worth recording your streaks, deleting your dailies, and putting
   them back to see if that fixes the problem.
    -19 days ago+1
   Cody Weston
   -

   @Luiza Mendonça Either that's damage from other party members, or it's a
   bug (a definite possibility).
    -19 days ago+1
   ferox
   -

   I don't understand, sometimes I have a perfect day, and I still get
   damage! That's the second or third time this happens... Do we get damage
   from undone to-dos???
    -19 days ago+1
   Luiza Mendonça
   -

   I use it since I signed up (and to be honest, i don't remember when that
   was!). I love it, I never drank so much water, took so many stairs or did
   so much exercise before in my life (my habits hehehe). I really love the
   social facet of it, it really helps you to stay motivated, and everyone is
   soooooo nice!!! I just wish you could put an agenda on it...
    -19 days ago+1
   Luiza Mendonça
   -

   Luiza Mendonça attacks Rat King for 4.9 damage, Rat King attacks party
   for 5.4 damage.
    -19 days ago+1

2014-08-11 15:52 GMT-07:00 deilann notifications@github.com:

> It sounds like you're not fetching your most recent party messages if you
> think the damage isn't showing for more than a day. You need to refresh
> the
> party page or ask for the new messages to see where the damage is coming
> from. Also, the rat king is not an easy quest. It is one of the hardest.
> Also remember that you take damage from both the boss and your dailies
> directly and you'll be taking damage from your party members.

**Comment by Alys** [comment]
@Luizammend : The messages in party chat are always chronological - they're stored in HabitRPG's database in the order they were created and they're reported on the party page in the exact order that the database has them. So the damage is being done at the right time, but what's happening is that your party chat is not being reloaded. Chats throughout HabitRPG don't automatically show new messsages; you have to sync to see them (sometimes a browser refresh works too). Each time you visit the party page, click on the circular arrow button that's next to the "Send Chat" button. That will always show you new messages, including your damage.

If you're taking more damage than you expect from your dailies, it's probably because sync errors are preventing your dailies from actually being ticked off, even if it looks like they are. Your browser knows you ticked the dailies, but it doesn't send that information to HabitRPG's server. This especially happens if you tick several dailies in quick succession, but can happen at other times too. Some tips:
- When you tick one task (or take any action), wait for the thin blue line at the top of the screen to finish moving to the right and then disappear.
- Just before you finish using HabitRPG for the night, press the sync button in the top right corner and wait for blue line to finish, and then check that all your dailies are still ticked. If not, tick them and sync again. You can also use the [User Data Display tool](https://oldgods.net/habitrpg/habitrpg_user_data_display.html) to see if you have any unchecked dailies (it's reliable because it takes data directly from Habit's servers).

**Comment by Luizammend** [comment]
Ok, I'll try doing that. What was really weird is that this occurred today,
in the middle of the day (I had entered HabitRPG already in the morning). I
know this damage was due to 2 specific dailies (one green and one blue)
because I have only two dailies that apply to weekends, and I didn't enter
habitRPG yesterday, so I know the damage came from those dailies only.
thank you,
Luiza

2014-08-11 16:59 GMT-07:00 Alice Harris notifications@github.com:
